pkg/app/piped/executor/waitapproval/waitapproval.go

```diff
@@ -161,3 +180,38 @@ func (e *Executor) getMentionedAccounts(event model.NotificationEventType) ([]st
 
 	return notification.FindSlackAccounts(event), nil
 }
+
+// An empty string means that number of approvers is invalid
+func (e *Executor) validateApproverNum(ctx context.Context, approver string, minApproverNum int) string {
+	if minApproverNum <= 1 {
+		e.LogPersister.Infof("Got approval from %q", approver)
+		return approver
+	}
+	as, ok := e.MetadataStore.Stage(e.Stage.Id).Get(approversKey)
+	if !ok {
+		e.LogPersister.Infof("Got approval from %q", approver)
+		e.LogPersister.Infof("Waiting for %d other approvers...", minApproverNum-1)
+		if err := e.MetadataStore.Stage(e.Stage.Id).Put(ctx, approversKey, approver); err != nil {
+			e.LogPersister.Errorf("Unable to save approver information to deployment, %v", err)
+		}
+		return ""
+	}
```

Comment:
Do we need this block? From my view, it is doing the same thing with the code below so we can simplify this part by

```
as, _ := e.MetadataStore.Stage(e.Stage.Id).Get(approversKey)
```

Reply:
Yes, we do. If we don't return here, it will be bug in `if c := minApproverNum - len(strings.Split(totalAs, ", ")); c > 0` because totalAs may be `["", "user-1"]`.

https://github.com/pipe-cd/pipe/pull/2802/files#diff-02a6821aa5f1210155af9610f8bb1b7683ed309d3ad56a533f0505d12ae6ffd6R208-R209